packages/headless/src/features/insight-user-actions/insight-user-actions-state.ts

```diff
@@ -15,15 +16,15 @@ export interface UserSession {
 export interface UserAction {
   actionType: UserActionType;
   timestamp: string;
-  eventData: {
-    type: string;
+  eventData?: {
+    type?: string;
```

Comment:
This cannot be optional? why is it?

Reply:
you mean the eventData? some events don't have a `event_type` or `event_value` thats why I marked it as optional. Only custom events seem to have those